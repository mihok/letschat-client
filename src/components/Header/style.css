/* Header Component */


/* Messenger Theme */

.Header--messenger {
  background: #72a0e5;
  box-sizing: border-box;
  color: #efefef;
  padding: 12px 4px 12px 12px;
  display: flex;
  justify-content: space-between;
  height: 32px;
  border-radius: 3px 3px 0 0;
  border-right: 1px solid rgba(0,0,0,0.1);
  cursor: pointer;
}

.Header__title--messenger {
  align-self: center;
  color: white;
}

.Header__closeBtn--messenger {
  font-weight: 800;
  background: none;
  border: 0;
  color: white;
  font-size: 18px;
  align-self: center;
  outline: none;
  cursor: pointer;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  line-height: 1;
  padding-top: 0;
}

.Header__closeBtn--messanger:active {
  box-shadow: 1px 1px 4px rgba(0, 0, 0, 0);
}


/* Float Theme */

.Header--float { }

.closed .Header--float {
  position: relative;
  right: 10px;
  bottom: 30px;
}

.Header__title--float {
  display: none;
}

.Header__closeBtn--float {
  position: absolute;
  bottom: 0;
  right: 0;
  display: block;
  width: 42px;
  height: 42px;
  padding: 4px 6px;
  box-sizing: border-box;
  box-shadow: #dddddd 1px 1px 8px 0;
  border-radius: 21px;
  font-size: 32px;
  font-family: sans-serif;
  color: #ffffff;
  cursor: pointer;
  outline: none;
  background-color: #72a0e5;
  border: 0;
}
