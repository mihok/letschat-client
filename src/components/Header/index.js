import { h, Component } from 'preact';
import PropTypes from 'prop-types';

import { applyTheme } from '../ThemeProvider/';

import './style.css';

/**
 * Header
 *
 */
class Header extends Component {
  propTypes = {
    toggleChat: PropTypes.func,
    style: PropTypes.string,
    theme: PropTypes.shape({
      primary_colour: PropTypes.string.isRequired,
    }),
    chatOpen: PropTypes.boolean,
  };

  renderToggleChatButton = () => (this.props.chatOpen ? <span>Ã—</span> : <span>...</span>);

  render () {
    const { toggleChat, style, theme } = this.props;

    return (
      <header
        style={{ backgroundColor: theme.primary_colour }}
        role="button"
        tabIndex={0}
        className={`Header--${style}`}
        onClick={() => toggleChat(true)}
        onKeyPress={event => {
          // Ensure event is not null
          const e = event || window.event;

          if ((e.which === 72 || e.keyCode === 72) && e.ctrlKey) {
            toggleChat(true);
          }
        }}
      >
        <span className={`Header__title--${style}`}>Chat with John</span>
        <button className={`Header__closeBtn--${style}`} onClick={() => toggleChat(true)}>
          {this.renderToggleChatButton()}
        </button>
      </header>
    );
  }
}

export default applyTheme(Header);
