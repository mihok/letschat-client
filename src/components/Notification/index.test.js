import { h } from 'preact';
import render from 'preact-render-to-string';

import Header from '.';

describe('<Header />', () => {
  it('should render ', () => {
    const tree = render(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
