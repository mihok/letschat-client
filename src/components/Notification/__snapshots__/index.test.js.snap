// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render  1`] = `"<header class=\\"Header--undefined\\"><span class=\\"Header__title--undefined\\">Chat with John</span><button class=\\"Header__closeBtn--undefined\\">Ã—</button></header>"`;
